src/libtls/tls_crypto.c

```diff
@@ -2425,3 +2425,60 @@ tls_named_group_t tls_ec_group_to_curve(diffie_hellman_group_t group)
 	}
 	return 0;
 }
+
+/*
+ * See header.
+ */
+hashtable_t *tls_get_server_key_types(tls_version_t min_version,
+									  tls_version_t max_version)
+{
+	hashtable_t *ht;
+	int i;
+
+	ht = hashtable_create((hashtable_hash_t) hashtable_hash_ptr,
+					   	  (hashtable_equals_t) hashtable_equals_ptr, 4);
```

Comment:
Not sure if this is the correct way to create hashes and compare them...

Reply:
No, this compares the pointers, not the values they point to. So you better write your own compare/hash functions (the latter usually use `chunk_hash[_inc]()`). You should find plenty of examples in other places.